:0: Unknown function 'Elixir.Aewallet.KeyPair':compress/1
:0: Unknown function 'Elixir.Aewallet.KeyPair':derive/2
:0: Unknown function 'Elixir.Aewallet.KeyPair':generate_master_key/1
:0: Unknown function 'Elixir.Aewallet.Signing':sign/2
:0: Unknown function 'Elixir.Aewallet.Signing':verify/3
:0: Unknown function 'Elixir.Aewallet.Wallet':create_wallet/2
:0: Unknown function 'Elixir.Aewallet.Wallet':get_private_key/3
:0: Unknown function 'Elixir.Aewallet.Wallet':get_public_key/3
:0: Unknown function 'Elixir.Aewallet.Wallet':get_seed/2
:0: Unknown function 'Elixir.ExJsonSchema.Schema':resolve/1
:0: Unknown function 'Elixir.ExJsonSchema.Validator':validate/2
:0: Unknown function 'Elixir.Msgpax':'pack!'/1
:0: Unknown function 'Elixir.Msgpax':'pack!'/2
:0: Unknown function 'Elixir.Msgpax':unpack/1
:0: Unknown function base58:base58_to_binary/1
:0: Unknown function base58:binary_to_base58/1
:0: Unknown function enacl:generichash/2
:0: Unknown function sha3:hash/2
:0: Unknown type 'Elixir.Account':t/0
:0: Unknown type 'Elixir.Aecore.Account.Account':pubkey/0
:0: Unknown type 'Elixir.Aecore.Chain.Chainstate':account_chainstate/0
:0: Unknown type 'Elixir.Aecore.Chain.Chainstate':accounts/0
:0: Unknown type 'Elixir.Aecore.Chain.Chainstate':chainstate/0
:0: Unknown type 'Elixir.Aecore.Chain.Chainstate':oracles/0
:0: Unknown type 'Elixir.Aecore.Structures.OracleExtendTx':t/0
:0: Unknown type 'Elixir.Aecore.Structures.OracleQueryTx':t/0
:0: Unknown type 'Elixir.Aecore.Structures.OracleRegistrationTx':t/0
:0: Unknown type 'Elixir.Aecore.Structures.OracleResponseTx':t/0
:0: Unknown type 'Elixir.Block':t/0
:0: Unknown type 'Elixir.ChainState':account_chainstate/0
:0: Unknown type 'Elixir.ChainState':accounts/0
:0: Unknown type 'Elixir.Chainstate':accounts/0
:0: Unknown type 'Elixir.List':t/0
:0: Unknown type 'Elixir.SignedTx':t/0
apps/aecore/lib/aecore/account/account.ex:101: The pattern {'ok', Vaccount_state@1} can never match the type binary() | #{'__struct__':='Elixir.Aecore.Account.Account', 'balance':=non_neg_integer(), 'last_updated':=non_neg_integer(), 'nonce':=non_neg_integer()}
apps/aecore/lib/aecore/account/tx/spend_tx.ex:108: The pattern Verr@1 = {'error', _reason@1} can never match the type 'ok'
apps/aecore/lib/aecore/chain/worker.ex:469: The call 'Elixir.Map':merge(Vblocks_info_map@2::'not_found' | {'error',_} | {'ok',map()},Vblocks_map@3::'not_found' | {'error',_} | {'ok',map()} | #{},fun((_,_,_) -> #{'block':=_, _=>_})) will never return since the success typing is (map(),map(),fun((_,_,_) -> any())) -> any() and the contract is (map(),map(),fun((key(),value(),value()) -> value())) -> map()
apps/aecore/lib/aecore/chain/worker.ex:472: The created fun has no local return
apps/aecore/lib/aecore/miner/worker.ex:77: Function mine_sync_block_to_chain/0 has no local return
apps/aecore/lib/aecore/miner/worker.ex:80: The call 'Elixir.Aecore.Miner.Worker':mine_sync_block(Vcblock@1::{'error',_}) will never return since the success typing is (#{'__struct__':='Elixir.Aecore.Chain.Block', _=>_}) -> {'error','miner_is_busy'} | {'ok',#{'__struct__':='Elixir.Aecore.Chain.Block', 'header':=map(), _=>_}} and the contract is ('Elixir.Aecore.Chain.Block':t()) -> {'ok','Elixir.Aecore.Chain.Block':t()} | {'error',reason::atom()}
apps/aecore/lib/aecore/miner/worker.ex:277: Function create_coinbase_tx/2 has no local return
apps/aecore/lib/aecore/miner/worker.ex:283: The call 'Elixir.Aecore.Tx.DataTx':init('Elixir.Aecore.Account.Tx.SpendTx',Vpayload@1::#{'amount':=number(), 'receiver':=_},'nil',0,0) breaks the contract (tx_types(),payload(),binary(),integer(),integer()) -> 'Elixir.Aecore.Tx.DataTx':t()
apps/aecore/lib/aecore/miner/worker.ex:304: The call 'Elixir.Aecore.Oracle.Oracle':'tx_ttl_is_valid?'(Vtx@1::#{'__struct__':='Elixir.Aecore.Tx.SignedTx', 'data':=#{'__struct__':='Elixir.Aecore.Tx.DataTx', 'fee':=non_neg_integer(), 'nonce':=non_neg_integer(), 'payload':=#{'__struct__':='Elixir.Aecore.Account.Tx.SpendTx' | 'Elixir.Aecore.Oracle.Tx.OracleExtendTx' | 'Elixir.Aecore.Oracle.Tx.OracleQueryTx' | 'Elixir.Aecore.Oracle.Tx.OracleRegistrationTx' | 'Elixir.Aecore.Oracle.Tx.OracleResponseTx', _=>_}, 'sender':=binary(), 'type':='Elixir.Aecore.Account.Tx.SpendTx' | 'Elixir.Aecore.Structures.OracleExtendTx' | 'Elixir.Aecore.Structures.OracleQueryTx' | 'Elixir.Aecore.Structures.OracleRegistrationTx' | 'Elixir.Aecore.Structures.OracleResponseTx'}, 'signature':=binary()},Vblock_height@1::pos_integer()) breaks the contract (oracle_txs_with_ttl(),non_neg_integer()) -> boolean()
apps/aecore/lib/aecore/miner/worker.ex:308: Function create_block/5 will never be called
apps/aecore/lib/aecore/oracle/oracle.ex:55: Function register/5 has no local return
apps/aecore/lib/aecore/oracle/oracle.ex:64: The call 'Elixir.Aecore.Tx.DataTx':init('Elixir.Aecore.Oracle.Tx.OracleRegistrationTx',Vpayload@1::#{'query_fee':=_, 'query_format':=_, 'response_format':=_, 'ttl':=_},binary(),Vfee@1::any(),non_neg_integer()) breaks the contract (tx_types(),payload(),binary(),integer(),integer()) -> 'Elixir.Aecore.Tx.DataTx':t()
apps/aecore/lib/aecore/oracle/oracle.ex:82: Function query/6 has no local return
apps/aecore/lib/aecore/oracle/oracle.ex:92: The call 'Elixir.Aecore.Tx.DataTx':init('Elixir.Aecore.Oracle.Tx.OracleQueryTx',Vpayload@1::#{'oracle_address':=_, 'query_data':=_, 'query_fee':=_, 'query_ttl':=_, 'response_ttl':=_},binary(),Vfee@1::any(),non_neg_integer()) breaks the contract (tx_types(),payload(),binary(),integer(),integer()) -> 'Elixir.Aecore.Tx.DataTx':t()
apps/aecore/lib/aecore/oracle/oracle.ex:109: Function respond/3 has no local return
apps/aecore/lib/aecore/oracle/oracle.ex:116: The call 'Elixir.Aecore.Tx.DataTx':init('Elixir.Aecore.Oracle.Tx.OracleResponseTx',Vpayload@1::#{'query_id':=_, 'response':=_},binary(),Vfee@1::any(),non_neg_integer()) breaks the contract (tx_types(),payload(),binary(),integer(),integer()) -> 'Elixir.Aecore.Tx.DataTx':t()
apps/aecore/lib/aecore/oracle/oracle.ex:129: Function extend/2 has no local return
apps/aecore/lib/aecore/oracle/oracle.ex:135: The call 'Elixir.Aecore.Tx.DataTx':init('Elixir.Aecore.Oracle.Tx.OracleExtendTx',Vpayload@1::#{'ttl':=_},binary(),Vfee@1::any(),non_neg_integer()) breaks the contract (tx_types(),payload(),binary(),integer(),integer()) -> 'Elixir.Aecore.Tx.DataTx':t()
apps/aecore/lib/aecore/oracle/tx/oracle_extend_tx.ex:41: Invalid type specification for function 'Elixir.Aecore.Oracle.Tx.OracleExtendTx':'process_chainstate!'/7. The success typing is (#{'__struct__':='Elixir.Aecore.Oracle.Tx.OracleExtendTx', 'ttl':=non_neg_integer(), _=>_},_,non_neg_integer(),_,_,map(),#{'registered_oracles':=map(), _=>_}) -> {map(),_}
apps/aecore/lib/aecore/oracle/tx/oracle_extend_tx.ex:50: The inferred type for the 6th argument of 'process_chainstate!'/7 (map()) is not a supertype of gb_merkle_trees:tree(), which is expected type for this argument in the callback of the 'Elixir.Aecore.Tx.Transaction' behaviour
apps/aecore/lib/aecore/oracle/tx/oracle_extend_tx.ex:50: The inferred return type of 'process_chainstate!'/7 ({map(),_}) has nothing in common with {'error' | gb_merkle_trees:tree(),binary() | map()}, which is the expected return type for the callback of the 'Elixir.Aecore.Tx.Transaction' behaviour
apps/aecore/lib/aecore/oracle/tx/oracle_query_tx.ex:83: The return type {'error' | gb_merkle_trees:tree(),_} in the specification of 'process_chainstate!'/7 is not a subtype of {'error' | gb_merkle_trees:tree(),binary() | map()}, which is the expected return type for the callback of the 'Elixir.Aecore.Tx.Transaction' behaviour
apps/aecore/lib/aecore/peers/sync.ex:142: The created fun has no local return
apps/aecore/lib/aecore/peers/sync.ex:292: The created fun has no local return
apps/aecore/lib/aecore/peers/sync.ex:359: The pattern 'error' can never match the type 'ok'
apps/aecore/lib/aecore/peers/worker.ex:100: The created fun has no local return
apps/aecore/lib/aecore/peers/worker.ex:132: The created fun has no local return
apps/aecore/lib/aecore/peers/worker.ex:175: The created fun has no local return
apps/aecore/lib/aecore/peers/worker.ex:283: Function trim_peers/1 will never be called
apps/aecore/lib/aecore/peers/worker.ex:297: Function check_peer/2 has no local return
apps/aecore/lib/aecore/peers/worker.ex:321: Function 'should_a_peer_be_added?'/1 will never be called
apps/aecore/lib/aecore/pow/cuckoo.ex:99: The call 'Elixir.Exexec':send(Vospid@1::non_neg_integer(),'eof') breaks the contract (pid() | os_pid(),binary()) -> 'ok'
apps/aecore/lib/aecore/pow/cuckoo.ex:163: Guard test '/='(Verror@1::'nil','nil') can never succeed
apps/aehttpclient/lib/client.ex:22: Function get_info/1 has no local return
apps/aehttpclient/lib/client.ex:27: Function get_block/1 has no local return
apps/aehttpclient/lib/client.ex:40: Function get_raw_blocks/1 has no local return
apps/aehttpclient/lib/client.ex:50: Function get_pool_txs/1 has no local return
apps/aehttpclient/lib/client.ex:55: Function send_block/2 has no local return
apps/aehttpclient/lib/client.ex:56: The call 'Elixir.Aeutil.Serialization':block(Vblock@1::any(),'serialize') will never return since it differs in the 2nd argument from the success typing arguments: (map(),'deserialize')
apps/aehttpclient/lib/client.ex:78: Function get_peers/1 has no local return
apps/aehttpclient/lib/client.ex:83: Function get_and_add_peers/1 has no local return
apps/aehttpclient/lib/client.ex:89: Function get_account_balance/1 has no local return
apps/aehttpclient/lib/client.ex:94: Function get_account_txs/1 has no local return
apps/aehttpclient/lib/client.ex:146: Function get/1 has no local return
apps/aehttpclient/lib/client.ex:146: Function get/2 has no local return
apps/aehttpserver/lib/aehttpserver/web/channels/notify.ex:22: Function broadcast_new_block_added_to_chain_and_new_mined_tx/1 has no local return
apps/aehttpserver/lib/aehttpserver/web/channels/notify.ex:32: The call 'Elixir.Aeutil.Serialization':block(Vblock@1::atom() | #{'txs':=_, _=>_},'serialize') will never return since it differs in the 2nd argument from the success typing arguments: (map(),'deserialize')
apps/aehttpserver/lib/aehttpserver/web/controllers/block_controller.ex:32: The call 'Elixir.Aeutil.Serialization':block(Vblock@1::any(),'serialize') will never return since it differs in the 2nd argument from the success typing arguments: (map(),'deserialize')
apps/aehttpserver/lib/aehttpserver/web/controllers/block_controller.ex:65: The created fun has no local return
apps/aehttpserver/lib/aehttpserver/web/controllers/block_controller.ex:70: The call 'Elixir.Aeutil.Serialization':block(Vblock@1::atom() | #{'header':=_, _=>_},'serialize') will never return since it differs in the 2nd argument from the success typing arguments: (map(),'deserialize')
apps/aehttpserver/lib/aehttpserver/web/controllers/block_controller.ex:110: The created fun has no local return
apps/aehttpserver/lib/aehttpserver/web/controllers/block_controller.ex:111: The call 'Elixir.Aeutil.Serialization':block(Vblock@1::any(),'serialize') will never return since it differs in the 2nd argument from the success typing arguments: (map(),'deserialize')
apps/aehttpserver/lib/aehttpserver/web/controllers/miner_controller.ex:15: The pattern {_, Vstate@1} can never match the type 'idle' | 'running'
apps/aehttpserver/lib/aehttpserver/web/controllers/oracle_controller.ex:12: Function oracle_response/2 has no local return
apps/aehttpserver/lib/aehttpserver/web/controllers/oracle_controller.ex:44: Function oracle_query/2 has no local return
apps/aehttpserver/lib/aehttpserver/web/controllers/tx_controller.ex:23: The call 'Elixir.Aeutil.Serialization':serialize_txs_info_to_json(Vuser_txs@1::[any(),...]) will never return since it differs in the 1st argument from the success typing arguments: ([])
apps/aehttpserver/lib/aehttpserver/web/controllers/tx_controller.ex:31: The call 'Elixir.Aeutil.Serialization':serialize_txs_info_to_json(Vuser_txs@1::[any(),...]) will never return since it differs in the 1st argument from the success typing arguments: ([])
apps/aeutil/lib/serialization.ex:35: The call 'Elixir.Access':get(Vserialized_block@1::binary(),#{#<116>(8, 1, 'integer', ['unsigned', 'big']), #<120>(8, 1, 'integer', ['unsigned', 'big']), #<115>(8, 1, 'integer', ['unsigned', 'big'])}#,'nil') will never return since the success typing is ('nil' | maybe_improper_list() | map(),any(),any()) -> any() and the contract is (container(),term(),term()) -> term()
    ; (nil_container(),any(),default) -> default
apps/aeutil/lib/serialization.ex:289: The call 'Elixir.Aecore.Tx.SignedTx':hash_tx(#{'__struct__':='Elixir.Aecore.Tx.SignedTx', 'data':=#{'__struct__':='Elixir.Aecore.Tx.DataTx', 'fee':=non_neg_integer(), 'nonce':=non_neg_integer(), 'payload':=_, 'sender':=binary(), 'type':='Elixir.Aecore.Account.Tx.SpendTx' | 'Elixir.Aecore.Structures.OracleExtendTx' | 'Elixir.Aecore.Structures.OracleQueryTx' | 'Elixir.Aecore.Structures.OracleRegistrationTx' | 'Elixir.Aecore.Structures.OracleResponseTx'}, 'signature':='nil'}) breaks the contract ('Elixir.Aecore.Tx.SignedTx':t()) -> binary()
